beginner_source/blitz/tensor_tutorial.py

```diff
@@ -1,195 +1,166 @@
-# .. note::
-#     ``torch.Size`` is in fact a tuple, so it supports all tuple operations.
-#
-# Operations
-# ^^^^^^^^^^
-# There are multiple syntaxes for operations. In the following
-# example, we will take a look at the addition operation.
+######################################################################
+# This works after we override the dtype property
 #
-# Addition: syntax 1
-y = torch.rand(5, 3)
-print(x + y)
 
-###############################################################
-# Addition: syntax 2
+new_rand_tnsr = torch.rand_like(tnsr_from_data, dtype=torch.float)
+print(new_rand_tnsr)
 
-print(torch.add(x, y))
 
-###############################################################
-# Addition: providing an output tensor as argument
-result = torch.empty(5, 3)
-torch.add(x, y, out=result)
-print(result)
 
-###############################################################
-# Addition: in-place
+######################################################################
+# --------------
+#
 
-# adds x to y
-y.add_(x)
-print(y)
 
-###############################################################
-# .. note::
-#     Any operation that mutates a tensor in-place is post-fixed with an ``_``.
-#     For example: ``x.copy_(y)``, ``x.t_()``, will change ``x``.
+######################################################################
+# Tensor Attributes
+# ~~~~~~~~~~~~~~~~~
+#
+# Tensors have attributes that describe their contents and functions. They
+# also have attributes that autograd uses to keep track of them in the
+# computational graph (more on this in the next section).
+#
+# **Docs Issues** - https://pytorch.org/docs/stable/tensor_attributes.html
+# is not comprehensive (missing data, grad, grad_fn, shape). Contains
+# ``memory_format`` which is not an attribute
 #
-# You can use standard NumPy-like indexing with all bells and whistles!
 
-print(x[:, 1])
+tnsr = torch.rand(3,4)
 
-###############################################################
-# Resizing: If you want to resize/reshape tensor, you can use ``torch.view``:
-x = torch.randn(4, 4)
-y = x.view(16)
-z = x.view(-1, 8)  # the size -1 is inferred from other dimensions
-print(x.size(), y.size(), z.size())
+print(f"Data stored in tensor:\n{tnsr.data}\n")
+print(f"Shape of tensor: {tnsr.shape}")
+print(f"Datatype of tensor: {tnsr.dtype}")
+print(f"Device tensor lives on: {tnsr.device}")
 
-###############################################################
-# If you have a one element tensor, use ``.item()`` to get the value as a
-# Python number
-x = torch.randn(1)
-print(x)
-print(x.item())
+print("-------")
 
-###############################################################
-# **Read later:**
-#
-#
-#   100+ Tensor operations, including transposing, indexing, slicing,
-#   mathematical operations, linear algebra, random numbers, etc.,
-#   are described
-#   `here <https://pytorch.org/docs/torch>`_.
+# Autograd-related attributes; more on this later
```

Comment:
I think these can wait until the autograd part no?

Reply:
Yes, removing from here